index.js

```diff
@@ -141,29 +156,57 @@ NYC.prototype.addAllFiles = function () {
   this.writeCoverageFile()
 }
 
-NYC.prototype._wrapRequire = function () {
+NYC.prototype._maybeInstrumentSource = function (code, filename, relFile) {
+  var instrument = this.shouldInstrumentFile(filename, relFile)
+
+  if (!instrument) {
+    return null
+  }
+
+  return this.transform(code, filename)
+}
+
+NYC.prototype._transformFactory = function (cacheDir) {
   var _this = this
-  appendTransform(function (code, filename) {
-    var relFile = path.relative(_this.cwd, filename)
-    var instrument = _this.shouldInstrumentFile(filename, relFile)
+  var instrumenter = this.instrumenter()
+
+  return function (code, filename, hash) {
+    var relFile = './' + path.relative(_this.cwd, filename)
+
+    var sourceMap = convertSourceMap.fromSource(code) || convertSourceMap.fromMapFileSource(code, path.dirname(filename))
 
-    if (!instrument) {
-      return code
+    if (hash) {
+      _this.hashCache[relFile] = hash
+      if (sourceMap) {
+        var mapPath = path.join(cacheDir, hash + '.map')
+        fs.writeFileSync(mapPath, sourceMap.toJSON())
+      }
+    } else if (sourceMap) {
+      _this.sourceMapCache.addMap(relFile, sourceMap.toJSON())
     }
 
-    _this.sourceMapCache.add(filename, code)
+    return instrumenter.instrumentSync(code, relFile)
+  }
+}
 
-    // now instrument the compiled code.
-    return _this.instrumenter.instrumentSync(code, './' + relFile)
+NYC.prototype._wrapRequire = function () {
+  var _this = this
+  appendTransform(function (code, filename) {
+    var relFile = path.relative(_this.cwd, filename)
+    return _this._maybeInstrumentSource(code, filename, relFile) || code
   })
 }
 
 NYC.prototype.cleanup = function () {
-  if (!process.env.NYC_CWD) rimraf.sync(this.tmpDirectory())
+  if (!process.env.NYC_CWD) rimraf.sync(this.tempDirectory())
 }
 
-NYC.prototype._createOutputDirectory = function () {
-  mkdirp.sync(this.tmpDirectory())
+NYC.prototype.clearCache = function () {
+  rimraf.sync(this.cacheDirectory())
+}
+
+NYC.prototype._createDatastoreDirectories = function () {
+  mkdirp.sync(this.tempDirectory())
```

Comment:
This now only creates the temp directory, since `cache-transform` takes care of creating the cache directory.